neuro_extras/image.py

```diff
@@ -155,6 +157,83 @@ def image_build(
                     preset=preset,
                     force_overwrite=force_overwrite,
                     verbose=verbose,
+                    local=False,
+                    build_tags=build_tag,
+                )
+            )
+        )
+    except (ValueError, click.ClickException) as e:
+        logger.error(f"Failed to build image: {e}")
+        sys.exit(EX_PLATFORMERROR)
+
+
+@image.command(
+    "local-build", help="Build Job container image locally (requires Docker daemon)."
+)
+@click.argument("path", metavar="CONTEXT_PATH")
+@click.argument("image_uri")
+@click.option(
+    "-f",
+    "--file",
+    default="Dockerfile",
+    show_default=True,
+    help=(
+        "Relative (w.r.t. context) path to the dockerfile. "
+        "The dockerfile should be within the context directory."
+    ),
+)
+@click.option(
+    "--build-arg",
+    multiple=True,
+    metavar="VAR=VAL",
+    help=(
+        "Build-time variables passed in ARG values. "
+        "Could be used multiple times for multiple arguments."
+    ),
+)
+@click.option(
+    "-F",
+    "--force-overwrite",
+    default=False,
+    show_default=True,
+    is_flag=True,
+    help="Overwrite if the destination image already exists.",
+)
+@click.option(
+    "--verbose",
+    type=bool,
+    default=False,
+    help="If specified, provide verbose output (default False).",
+)
+@click.option(
+    "--build-tag",
+    multiple=True,
+    metavar="VAR=VAL",
+    help=("Set tag(s) for image builder job. "),
+)
+def image_build_local(
+    path: str,
+    image_uri: str,
+    file: str,
+    build_arg: Tuple[str],
+    force_overwrite: bool,
+    verbose: bool,
+    build_tag: Tuple[str],
+) -> None:
+    try:
+        sys.exit(
+            asyncio.run(
+                _build_image(
+                    dockerfile_path=Path(file),
+                    context=path,
+                    image_uri_str=image_uri,
+                    use_cache=True,
```

Comment:
Actually, we could also enable or disable caching in local builds.
The caching is available and enabled if the [docker-buildkit](https://pythonspeed.com/articles/docker-buildkit/) is used (and it is used by default since 2020's releases).
This is optional, but we could set `DOCKER_BUILDKIT=0` for local builds if the caching is disabled.

Reply:
Ok, but I don't see any good reason to expose this functionality via the CLI. If needed, users can set the env var by hand.

Reply:
True, exposing this via caching parameter is a bit confusing. Let's leave it as is for now. Thank you!